Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -550,6 +579,89 @@ namespace GradientSignal
         SetupMultiplierAndOffset(m_configuration.m_scaleRangeMin, m_configuration.m_scaleRangeMax);
     }
 
+    float ImageGradientComponent::GetValueForSamplingType(AZ::u32 x0, AZ::u32 y0, float pixelX, float pixelY) const
+    {
+        switch (m_currentSamplingType)
+        {
+        case SamplingType::Point:
+        default:
+            // Retrieve the pixel value for the single point
+            return GetPixelValue(x0, y0);
+
+        case SamplingType::Bilinear:
+            // Bilinear interpolation
+            //
+            //   |
+            //   |
+            //   |  (x0,y1) *             * (x1,y1)
+            //   |
+            //   |                o (x,y)
+            //   |
+            //   |  (x0,y0) *             * (x1,y0)
+            //   |___________________________________
+            //
+            // The bilinear filtering samples from a grid around a desired pixel (x,y)
+            // x0,y0 contains one corner of our grid square, x1,y1 contains the opposite corner, and deltaX/Y is the fractional
+            // amount the position exists between those corners.
+            // Ex: (3.3, 4.4) would have a x0,y0 of (3, 4), a x1,y1 of (4, 5), and a deltaX/Y of (0.3, 0.4).
+            const auto& width = m_imageDescriptor.m_size.m_width;
+            const auto& height = m_imageDescriptor.m_size.m_height;
+            float deltaX = pixelX - floor(pixelX);
+            float deltaY = pixelY - floor(pixelY);
+
+            // Handle the x1,y1 points based on the configured wrapping type
+            AZ::u32 x1;
+            AZ::u32 y1;
+            bool x1IsValid = true;
+            bool y1IsValid = true;
+            switch (m_gradientTransform.GetWrappingType())
+            {
+            case WrappingType::ClampToEdge:
+                x1 = AZStd::min(x0 + 1, width - 1);
+                y1 = AZStd::min(y0 + 1, height - 1);
+                break;
+            case WrappingType::ClampToZero:
+                // For ClampToZero, the value will always be 0 outside of the shape
+                // So go ahead and do the calculation here, but if it ends
+                // up being outside of the shape, then it will be considered
+                // invalid and we will use 0 for the value below
+                x1 = x0 + 1;
+                if (x1 >= width)
+                {
+                    x1IsValid = false;
+                }
+                y1 = y0 + 1;
+                if (y1 >= height)
+                {
+                    y1IsValid = false;
+                }
+            case WrappingType::Mirror:
```

Comment:
The ClampToZero case appears to be missing a "break;"

Reply:
Not "appears to be", definitely is, good catch I will fix :)